internal/agentdeployer/factory.go

```diff
@@ -80,23 +67,51 @@ func Factory(options FactoryOptions) (AgentDeployer, error) {
 		// FIXME: this docker-compose scenario contains both agent and service
 		return nil, nil
 	case "k8s":
-		if _, err := os.Stat(agentDeployerPath); err == nil {
-			opts := KubernetesAgentDeployerOptions{
-				Profile:        options.Profile,
-				DefinitionsDir: agentDeployerPath,
-				StackVersion:   options.StackVersion,
-				PolicyName:     options.PolicyName,
-				DataStream:     options.DataStream,
-				RunSetup:       options.RunSetup,
-				RunTestsOnly:   options.RunTestsOnly,
-				RunTearDown:    options.RunTearDown,
-			}
-			return NewKubernetesAgentDeployer(opts)
+		opts := KubernetesAgentDeployerOptions{
+			Profile:        options.Profile,
+			DefinitionsDir: agentDeployerPath,
+			StackVersion:   options.StackVersion,
+			PolicyName:     options.PolicyName,
+			DataStream:     options.DataStream,
+			RunSetup:       options.RunSetup,
+			RunTestsOnly:   options.RunTestsOnly,
+			RunTearDown:    options.RunTearDown,
 		}
+		return NewKubernetesAgentDeployer(opts)
 	}
 	return nil, fmt.Errorf("unsupported agent deployer (name: %s)", agentDeployerName)
 }
 
+func selectAgentDeployerType(options FactoryOptions) (string, string, error) {
+	devDeployPath, err := FindDevDeployPath(options)
+	if errors.Is(err, os.ErrNotExist) {
+		return "default", "", nil
+	}
+	if err != nil {
+		return "", "", fmt.Errorf("can't find \"%s\" directory: %w", options.DevDeployDir, err)
+	}
+
+	agentDeployerName, err := findAgentDeployer(devDeployPath)
+	if err != nil {
+		logger.Debugf("Not found any agent deployer, using default one")
+		return "default", "", nil
+	}
+	// if package defines `_dev/deploy/docker` folder to start their services, it should be
+	// using the default agent deployer`
+	if agentDeployerName == "docker" || agentDeployerName == "tf" {
+		return "default", "", nil
+	}
+
+	agentDeployerPath := filepath.Join(devDeployPath, agentDeployerName)
+	if info, err := os.Stat(agentDeployerPath); err != nil {
+		return "", "", fmt.Errorf("could not find deployer path for agent deployer %q: %w", agentDeployerName, err)
+	} else if errors.Is(err, os.ErrNotExist) || !info.IsDir() {
+		return "", "", fmt.Errorf("expected to find a directory in %s", agentDeployerPath)
+	}
```

Comment:
In case, this file path does not exist I think it is not going to execute the `else` block since `err != nil`.

Should this condition be added inthe first `if` ?

```suggestion
	if info, err := os.Stat(agentDeployerPath); err != nil && !errors.Is(err, os.ErrNotExist) {
		return "", "", fmt.Errorf("could not find deployer path for agent deployer %q: %w", agentDeployerName, err)
	} else if errors.Is(err, os.ErrNotExist) || !info.IsDir() {
		return "", "", fmt.Errorf("expected to find a directory in %s", agentDeployerPath)
	}
```